daemon/protocol/python/scoot/client_lib.py

```diff
@@ -90,13 +93,53 @@ def start():
     raise ScootException(Exception("Could not find domain socket path."))
   if is_started():
     raise ScootException(Exception("Already started."))
-  channel = grpc.insecure_channel("unix://%s" % _domain_sock)
+  channel = None
+  try:
+    channel = grpc.insecure_channel("unix://%s" % _domain_sock)
+  except Exception as e:
+    raise ScootException("Error getting channel:'unix://{}'".format(_domain_sock))
   #channel.subscribe(lambda x: sys.stdout.write(str(x)+"\n"), try_to_connect=True)
   _client = daemon_pb2.ScootDaemonStub(channel)
+  
+  # verify the connection
+  echoReturn = None
+  if _client is not None:
+    ping = str(uuid.uuid4())
+    try:
+      echoReturn = echo(ping=ping)
+    except ScootException as e:
+      if "UNAVAILABLE" in str(e):
+        echoReturn = str(e)
+      else:
+        raise ScootException("ScootException verifying connection: echo:'{}', '{}'". format(ping,e))
+    except Exception as e:
+        raise ScootException("Exception verifying connection: echo:'{}', '{}'". format(ping,e))
+      
+          
+  if _client is None or "UNAVAILABLE" in echoReturn:
+    #the echo failed, try starting the daemon 
+    gopath = re.split(":", os.environ['GOPATH'])[0]
+    daemon = gopath + '/bin/daemon'
+    subprocess.Popen([daemon,"-execer_type", "os"])
+    start = time.time()
+      
+    # give the daemon up to 3 seconds to start
+    while True:
+      if _client is None: # try to create a connection
+        _client = daemon_pb2.ScootDaemonStub(channel)
+      # send an echo request
+      ping = str(uuid.uuid4())
+      echoReturn = echo(ping=ping)
+      if ping in echoReturn:
+        break
+        
+      now = time.time()
+      if now - start > 3.0: #wait up to 3 seconds for the daemon to start
```

Comment:
some runs needed > 1 second - so wanted to avoid false negatives. 3 seemed a good choice.